src/Optimizely/DecisionService/DecisionService.php

```diff
@@ -148,6 +150,153 @@ public function getVariation(Experiment $experiment, $userId, $attributes = null
+    foreach($experimentIds as $experiment_id){
+      $experiment = $this->_projectConfig->getExperimentFromId($experiment_id);
+      if( $experiment == new Experiment()){
+        continue;
+      }
+
+      $variation = $this->getVariation($experiment, $userId, $userAttributes);
+      if($variation instanceof Variation && $variation != new Variation){
+        $this->_logger->log(Logger::INFO,
+          "The user '{$userId}' is bucketed into experiment '{$experiment->getKey()}' of feature '{$feature_flag_key}'.");
+        return array(
+          "experiment"=> $experiment,
+          "variation" => $variation
+        );
+      }
+    }
+
+    $this->_logger->log(Logger::INFO,
+      "The user '{$userId}' is not bucketed into any of the experiments on the feature '{$feature_flag_key}'.");
+
+    return null;
+  }
+
+   /**
+   * Get the variation if the user is bucketed for one of the rollouts on this feature flag
+   * Evaluate the user for rules in priority order by seeing if the user satisfies the audience.
+   * Fall back onto the everyone else rule if the user is ever excluded from a rule due to traffic allocation.
+   * @param  FeatureFlag $featureFlag The feature flag the user wants to access
+   * @param  string      $userId      user id
+   * @param  array       $userAttributes  user userAttributes
+   * @return Variation/null
+   */
+  public function  getVariationForFeatureRollout(FeatureFlag $featureFlag, $userId, $userAttributes){
+    $feature_flag_key = $featureFlag->getKey();
+    $rollout_id = $featureFlag->getRolloutId();
+    if(empty($rollout_id)){
+       $this->_logger->log(Logger::DEBUG,
+        "Feature flag '{$feature_flag_key}' is not used in a rollout.");
+       return null; 
+    }
+    $rollout = $this->_projectConfig->getRolloutFromId($rollout_id);
+    if($rollout == new Rollout()){
+      // Error logged and thrown in getRolloutFromId
+      return null;
+    }
+
+    $rolloutRules = $rollout->getExperiments();
+    if(sizeof($rolloutRules) == 0)
+      return null;
+
+    // Evaluate all rollout rules except for last one
+    for($i=0; $i<sizeof($rolloutRules)-1; $i++){
+      $experiment = $rolloutRules[$i];
+
+      // Evaluate if user meets the audience condition of this rollout rule
+      if (!Validator::isUserInExperiment($this->_projectConfig, $experiment, $userAttributes)) {
+        $this->_logger->log(
+          Logger::DEBUG,
+          sprintf("User '%s' did not meet the audience conditions to be in rollout rule '%s'.", $userId, $experiment->getKey())
+        );
+        // Evaluate this user for the next rule
+        continue;
+      }
+
+      $this->_logger->log(Logger::DEBUG,
+        sprintf("Attempting to bucket user '{$userId}' into rollout rule '%s'.", $experiment->getKey()));
+
+      // Evaluate if user satisfies the traffic allocation for this rollout rule
+      $variation = $this->_bucketer->bucket($this->_projectConfig, $experiment, $userId, $userId);
```

Comment:
Owais, bucketID shouldn't be userId, can you please check we get bucketing ID from the userAttributes and we get the userAttributes. Please revise this code.